cmd/booster-http/server.go

```diff
@@ -164,6 +187,27 @@ func (s *HttpServer) handleByPieceCid(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	if s.nitroRpcClient != nil {
+		params, _ := url.ParseQuery(r.URL.RawQuery)
+		if !params.Has("channelId") {
+			writeError(w, r, http.StatusPaymentRequired, "a valid channel id must be provided")
+			return
+		}
+		rawChId := params.Get("channelId")
+		chId := types.Destination(common.HexToHash(rawChId))
+
+		if (chId == types.Destination{}) {
+			writeError(w, r, http.StatusPaymentRequired, "a valid channel id must be provided")
+			return
+		}
+		// TODO: Allow this to be configurable
+		expectedPaymentAmount := big.NewInt(10)
+
+		if hasPaid := checkPaymentChannelBalance(s.nitroRpcClient, chId, expectedPaymentAmount); !hasPaid {
+			writeError(w, r, http.StatusPaymentRequired, "payment required")
```

Comment:
Probably out of scope for this PR, but I think with this model, one payment would unlock an unlimited number of retrievals. Longer term we need a better model.

Reply:
Yup! This doesn't check if you paid for a file; it checks that you've paid at least 10 on some channel (once).